app/src/main/java/com/nextcloud/talk/utils/ApiUtils.java

```diff
@@ -100,41 +105,102 @@ public static RetrofitBucket getRetrofitBucketForContactsSearchFor14(String base
         return retrofitBucket;
     }
 
+    public static String getUrlForCapabilities(String baseUrl) {
+        return baseUrl + ocsApiVersion + "/cloud/capabilities";
+    }
 
-    public static String getUrlForSettingNotificationlevel(String baseUrl, String token) {
-        return getRoom(baseUrl, token) + "/notify";
+    public static Integer getApiVersion(UserEntity capabilities, String apiName, int[] versions) {
+         if (apiName.equals("conversation")) {
+             boolean hasApiV4 = false;
+             for (int version : versions) {
+                 hasApiV4 |= version == 4;
+             }
+
+             if (!hasApiV4) {
+                 Exception e = new Exception("Api call did not try conversation-v4 api");
+                 Log.d(TAG, e.getMessage(), e);
```

Comment:
No need to create an exception just to log it in the next line. Also exceptions tend to be expensive especially when you throw them.